boards/native/Makefile.dep

```diff
@@ -1,5 +1,7 @@
 ifneq (,$(filter netdev_default,$(USEMODULE)))
-  USEMODULE += netdev_tap
+  ifeq (,$(filter socket_zep,$(USEMODULE)))
+    USEMODULE += netdev_tap
+  endif
```

Comment:
As far as I can see, this just makes things more complicated: Instead of having to include only `netdev_default` + `socket_zep`, we now need to include `netdev_default` (since we do that in the network applications for all boards) `netdev_tap` and `socket_zep`, for all applications that use networking with `native`. I prefer to keep this as is and thus not requiring no changes to the makefiles

Reply:
But when I want to simulate a 802.15.4 node on native, I don't want it to also have a `netdev_tap` interface

Reply:
But then you pick `socket_zep`. You have to pick `socket_zep` with this approach as well.

Reply:
So remove `netdev_default` and add `socket_zep` instead?
Yea I can do that.

Mind you that this behavior of 'overwriting' the default netdev is already used by [nrf52](https://github.com/RIOT-OS/RIOT/blob/master/boards/common/nrf52/nrf52840/Makefile.dep) and [esp*](https://github.com/RIOT-OS/RIOT/blob/master/cpu/esp_common/Makefile.dep#L37), I thought it would be straightforward to do the same for `native`.

Reply:
> Mind you that this behavior of 'overwriting' the default netdev is already used by [nrf52](https://github.com/RIOT-OS/RIOT/blob/master/boards/common/nrf52/nrf52840/Makefile.dep) and [esp*](https://github.com/RIOT-OS/RIOT/blob/master/cpu/esp_common/Makefile.dep#L37), I thought it would be straightforward to do the same for `native`.

I know, but in this instance, for this specific PR, it seems making more fuzz then necessary. If you want, we can discuss this in a separate PR. I agree this should be more unified to other platforms, but as we already learned, this is affecting other applications than the one targeted by this PR, so we should rather discuss it in a different scope.

Reply:
#15430

Reply:
Can this PR still be merged first?

Reply:
(with the changes of #15430 removed first of course)

Reply:
If you prefer it that way - I thought it would be easier to merge the smaller one first

Reply:
I think the smaller one is the bigger change, so I would like to get this one out of the way first and then discuss #15430.